### PR TITLE
[Feat]: Rerouted Typeform get form(s) endpoints to Mongo DB

### PR DESCRIPTION
**Pull request by elimanzo**
- feat(form routes): Added Mongo API call to get all forms and seperated Typeform Logic from Mongo API calls
- feat(routing): API routes to mongo for fetching forms and forms by id
- feat(responses): Added fetch form fields through our mongo db endpoint

### Description

Re-routed some of the typeform calls to our mongo db calls

### Issue Link

[Jira Link](https://cascarita.atlassian.net/browse/C1-99)

### Checklist

- [x] I have tested the changes locally by running `npm run test`
- [ ] I have added or updated relevant documentation
- [x] I have autoformatted the code by running `npm run eslint`
- [ ] I have added test cases (if applicable)
###  Extra Notes:
- **The Backend does need a lot of refactor, as there's a a lot usage of  `Schema.Types.Mixed` being used, and fetching the data is creating a lot of bugs when we don't know what data we're receiving (currently fetching form by id is returning an array instead of the form itself). I do assume these will get changed when we move away from typeform's api calls 😃 .**

## Breaking Changes
- Since we do not have Update/Delete endpoints, the Delete / New Form Save will be broken
- I was going to implement get Responses but this API has bugs currently, so I resorted to using Typeform responses API call. #82 is the issue.

### CODE REVIEW

**Review by abanuelo — CHANGES_REQUESTED** [review]
Great work. Just a couple of questions!

File: client/pages/Forms/Forms.tsx [hunk 1]
```diff
@@ -31,19 +36,26 @@ const Forms = () => {
...
         id,
-        title: form.title,
-        description: form.welcome_screens?.[0]?.properties?.description ?? "",
-        link: form._links.display,
-        fields: form.fields,
+        title: form[0].form_data.title,
+        description:
+          form[0].form_data.welcome_screens?.[0]?.properties?.description ?? "",
+        link: form[0].form_data._links.display,
+        fields: form[0].form_data.fields,
```
[the remark this answers is not in the document]
Comment (elimanzo) [reply client/pages/Forms/Forms.tsx:58]: Yeah I had the same question, the indexing for form, I wrote a comment about it, and that is what our back-end is currently fetching, i did mention that we shouldn't retrieve a collection of forms when we're trying to fetch only one

File: client/api/forms/service.ts [hunk 2]
```diff
@@ -107,36 +126,65 @@ export const updateForm = async (
...
 
     if (!response.ok) {
-      throw new Error("Failed to update form");
+      throw new Error(`Error fetching forms: ${response.statusText}`);
     }
 
     return response.json();
   } catch (error) {
-    console.error(error);
+    console.error("Error fetching forms:", error);
```
[the remark this answers is not in the document]
Comment (elimanzo) [reply client/api/forms/service.ts:157]: good catch! Will apply errors to throw + console logging them to have consistency :)

**Review by abanuelo — APPROVED** [review]
Thanks for changes!

**Pull request merged**